### DISCUSSION TITLE
Error in mvn install

### DISCUSSION THREAD

**Discussion by NorbertDN** [GSI-CS-CO/chart-fx#548] [Q&A]
Hello,

I have a problem when I run the "mvn install" command. I get the following error:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-plugin-plugin:3.6.4:descriptor (earlierPluginXML) on project chartfx-generate: Error extracting plugin descriptor: 'No mojo definitions were found for plugin: de.gsi.generate:chartfx-generate.' -> [Help 1] org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-plugin-plugin:3.6.4:descriptor (earlierPluginXML) on project chartfx-generate: Error extracting plugin descriptor: 'No mojo definitions were found for plugin: de.gsi.gene rate:chartfx-generate.' 
```

I tried all the solutions I found on the internet without it solving my problem. Do you know a way around this?

Thank you for your answer and for your development,

Norbert

**Comment by NorbertDN** [comment]
Hello,

After searching on the internet, I added the following lines to the pom.xml of the Chartfx-generate project:

            <configuration>
                       
                      <skipErrorNoDescriptorsFound>true</skipErrorNoDescriptorsFound>
		</configuration> 

and the project is ok:
`[INFO] chartfx-generate ................................... SUCCESS [  3.657 s]`
but it is the following project that does not pass:
`[INFO] dataset ............................................ FAILURE [  0.000 s]`
with the error:
`[ERROR] Could not find goal 'generate-sources' in plugin de.gsi.generate:chartfx-generate:11.2.7 among available goals -> [Help 1]
org.apache.maven.plugin.MojoNotFoundException: Could not find goal 'generate-sources' in plugin de.gsi.generate:chartfx-generate:11.2.7 among available goals`

How to compile the Chartx-dataset project?

Thank you for your reply,

Norbert

**Reply by wirew0rm** [reply]
If you skip generating the plugin descriptor it will fail to execute the plugin in the next step, so this is a dead-end.

**Comment by clisztian** [comment]
Yeah, it's really a shame. I've been having the same exact issue, and haven't been able to build it since their migration to fair-acc a couple months ago. I tried building in mvn in command, building in IntellIJ using maven, a maven build in Eclipse and haven't found a way yet. Just using an old build until the developers come back to address this. 
If you find a way, would be great to know.

**Reply by NorbertDN** [reply]
Hello,
Thank you for your reply. 
Is there any other way to get the latest version than using maven? 
Otherwise it means that only the developers have it.

**Reply by wirew0rm** [reply]
If you only want to use it and not make modifications to the library, you can always use the snapshot repository: https://github.com/fair-acc/chart-fx#using-the-snapshot-repository to get builds for the current main and all pull request branches.

Since we try to use final as little as possible, you can also override most functionality in your own code by subclassing specific classes.

But of course you should also be able to locally build the code. Since we require our CI to pass before merging PRs, the code in `main` always builds at least in the CI and the Developer's machine, so I suggest to compare your setup to the CI runners. Unfortunately, the amount of system configurations is too high for us to test or debug for every possible setup, but we are open for contributions that allow more people to build chart-fx locally.

**Comment by wirew0rm** [comment]
This is probably due to some older artefacts of chart-fx still in your .m2 repository or target directories which get incorrectly resolved, since de.gsi.generate is not mentioned anywhere in the repo at main.

Please try with a fresh clone and an empty .m2: `mkdir /tmp/.m2 && mvn -Dmaven.repo.local=/tmp/.m2/ compile`

Another issue could be the maven version, I'm using 3.8.5 (just as a reference, i don't know how old is too old).

If you are building from an IDE, some of them cannot load maven plugins from a maven module or have other quirks, so always try to build on the command line first.
